code-challenges.js

```diff
@@ -54,22 +121,103 @@ var refactorTester1 = true
 var refactorTester2 = 42
 var refactorTester3 = "IAmACodingMaster"
 
-
-
-
+// ***psudo***
+// refactor to look for different data types 
+
+
+// const onlyVowelsRefactor = (data) => {
+//   if(typeof data !== 'string'){
+//     return `${data} is not a string`
+//   }else {
+//     let strArr = data.split('')
+//     return strArr.filter(value => {
+//       return value !== 'a' && value !== 'e' && value !== 'i' && value !== 'o' && value !== 'u' 
+//     }).join('')
+//   }
+// }
+
+// **refactor for capital vowels**
+// const onlyVowelsRefactor = (data) => {
+//   if(typeof data !== 'string'){
+//     return `${data} is not a string`
+//   }else {
+//     let strArr = data.split('')
+//     return strArr.filter(value => {
+//       return value.toLocaleLowerCase() !== 'a' && value.toLocaleLowerCase() !== 'e' && value.toLocaleLowerCase() !== 'i' && value.toLocaleLowerCase() !== 'o' && value.toLocaleLowerCase() !== 'u' 
+//     }).join('')
+//   }
+// }
+
+// **refactor for cleaner code**
+const onlyVowelsRefactor = (data) => {
+  if(typeof data !== 'string'){
+    return `${data} is not a string`
+  }else {
+    let strArr = data.split('')
+    return strArr.filter(value => {
+      let lower = value.toLowerCase()
+      return lower !== 'a' && lower !== 'e' && lower !== 'i' && lower !== 'o' && lower !== 'u' 
+    }).join('')
+  }
+}
+
+
+console.log('Q5:',onlyVowelsRefactor(refactorTester1))
+console.log('Q5:',onlyVowelsRefactor(refactorTester2))
+console.log('Q5:',onlyVowelsRefactor(refactorTester3))
 
 // --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
 // Expected output: "chopsticks" "stemware"
 
 var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
 var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
 
-
-
-
+// ***psudo***
+//function that takes an array of strings
+//iterate through array 
+//find string with most caracters
+//output string with most caracters
+
+const mostChars = (arr) => {
+  let most = 0
```

Comment:
to help make your code easier/faster to read, variable names can be use to indicate the purpose of the variable